File: src/main/resources/zzz/src/services/workOrderService.js
// src/services/workOrderService.js
import api from './api'

export const workOrderService = {
  // 获取我的工单
  async getMyOrders(repairmanId) {
    try {
      const response = await api.get(`/api/app/repairman/my-orders?repairmanId=${repairmanId}`)
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  // 获取可抢工单
  async getAvailableOrders(areaId) {
    try {
      const response = await api.get(`/api/app/repairman/available-orders?areaId=${areaId}`)
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  // 抢单
  async grabOrder(orderId, repairmanId) {
    try {
      const response = await api.post('/api/app/repairman/grab-order', {
        orderId,
        repairmanId
      })
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  // 拒单
  async rejectOrder(orderId, repairmanId, reason) {
    try {
      const response = await api.post('/api/app/repairman/reject-order', {
        orderId,
        repairmanId,
        reason
      })
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  // 提交维修结果
  async submitRepairResult(orderId, repairmanId, dealNote, imgUrl = null) {
    try {
      const response = await api.post('/api/app/repairman/submit-result', {
        orderId,
        repairmanId,
        dealNote,
        imgUrl
      })
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  }
}

File: src/main/java/com/campus/water/service/WorkOrderService.java
package com.campus.water.service;

import com.campus.water.entity.WorkOrder;
import java.util.List;

public interface WorkOrderService {
    //按ID获取工单详情
    WorkOrder getOrderDetail(String orderId);

    // 抢单
    boolean grabOrder(String orderId, String repairmanId);

    // 拒单
    boolean rejectOrder(String orderId, String repairmanId, String reason);

    // 提交维修结果
    boolean submitRepairResult(String orderId, String repairmanId, String dealNote, String imgUrl);

    // 获取可抢工单
    List<WorkOrder> getAvailableOrders(String areaId);

    // 获取我的工单
    List<WorkOrder> getMyOrders(String repairmanId);

    // 管理员手动派单（新增方法）
    boolean assignOrderByAdmin(String orderId, String repairmanId);

    // 新增：审核工单方法（与实现类签名一致）
    boolean reviewOrder(String orderId, boolean approved);

    // 按状态查询工单的方法
    List<WorkOrder> getOrdersByStatus(WorkOrder.OrderStatus status);
    List<WorkOrder> getOrdersByAreaAndStatus(String areaId, WorkOrder.OrderStatus status);
}